core/vendor/innerHTML.js

```diff
@@ -0,0 +1,93 @@
+// see https://github.com/WebReflection/document-register-element/issues/21#issuecomment-102020311
+var innerHTML = (function (document) {
+
+  var
+    EXTENDS = 'extends',
+    register = document.registerElement,
+    div = document.createElement('div'),
+    dre = 'document-register-element',
+    innerHTML = register.innerHTML,
+    initialize,
+    registered
+  ;
+
+  // avoid duplicated wrappers
+  if (innerHTML) return innerHTML;
+
+  try {
+
+    // feature detect the problem
+    register.call(
+      document,
+      dre,
+      {prototype: Object.create(
+        HTMLElement.prototype,
+        {createdCallback: {value: Object}}
+      )}
+    );
+
+    div.innerHTML = '<' + dre + '></' + dre + '>';
+
+    // if natively supported, nothing to do
+    if ('createdCallback' in div.querySelector(dre)) {
+      // return just an innerHTML wrap
+      return (register.innerHTML = function (el, html) {
+        el.innerHTML = html;
+        return el;
+      });
+    }
+
+  } catch(meh) {}
+
+  // in other cases
+  registered = [];
+  initialize = function (el) {
+    if (
+      'createdCallback' in el         ||
+      'attachedCallback' in el        ||
+      'detachedCallback' in el        ||
```

Comment:
@anatoo The v1 polyfill maps `connectedCallback` -> `attachedCallback` so the helper still works even if it references `attachedCallback`:

![image](https://cloud.githubusercontent.com/assets/6549462/18505176/11e7abc0-7aa1-11e6-8b6b-39af15f17ac2.png)